File: app/src/main/java/com/app/catsmoker/RootLSPosedActivity.java
package com.app.catsmoker;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class RootLSPosedActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_root_lsposed);

        // Instructions TextView
        TextView instructions = findViewById(R.id.instructions);

        String instructionText = """
                Root & LSPosed Instructions:
                
                1. Check Root Access
                   - Ensure device is rooted
                   - Use 'Root Checker' to verify
                
                2. Install Magisk
                   - Download Magisk Canary
                
                3. (Optional) Install Shamiko Module
                   - Hides root detection
                
                4. Install LSPosed_mod Module
                   - Open Magisk and install
                
                5. Launch LSPosed Manager
                   - Open the app
                
                6. Enable the CatSmoker Module
                   - Modules → Search CatSmoker → Enable
                
                7. Manage Supported Games
                   - Supported games spoofed as OnePlus 12:
                
                8. Force Stop the Game
                   - Apply changes via LSPosed""";

        instructions.setText(instructionText);

        // Status TextView
        TextView status = findViewById(R.id.status);
        boolean isRooted = checkRootStatus(); // Placeholder for actual root check
        status.setText(isRooted ? "Activated" : "Disabled");

        // Button to refresh status or trigger manual check
        Button refreshButton = findViewById(R.id.btn_refresh);
        refreshButton.setOnClickListener(v -> {
            boolean updatedRootStatus = checkRootStatus();
            status.setText(updatedRootStatus ? "Activated" : "Disabled");
            Toast.makeText(this, "Status refreshed", Toast.LENGTH_SHORT).show();
        });

        // Button to install LSPosed
        Button installLSPosedButton = findViewById(R.id.btn_install_lsposed);
        installLSPosedButton.setOnClickListener(v -> {
            // Open a browser or download manager to install LSPosed
            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/LSPosed/LSPosed/releases"));
            startActivity(browserIntent);
        });

        // Button to launch LSPosed Manager
        Button launchLSPosedButton = findViewById(R.id.btn_launch_lsposed);
        launchLSPosedButton.setOnClickListener(v -> {
            // Attempt to launch LSPosed Manager
            Intent intent = getPackageManager().getLaunchIntentForPackage("org.lsposed.manager");
            if (intent != null) {
                startActivity(intent);
            } else {
                Toast.makeText(this, "LSPosed Manager not installed", Toast.LENGTH_SHORT).show();
            }
        });
    }

    // Placeholder method for root checking - implement actual logic as needed
    private boolean checkRootStatus() {
        // This is a simulation. In a real app, use a root detection library like RootBeer
        try {
            Runtime.getRuntime().exec("su");
            return true;
        } catch (Exception e) {
            return false;
        }
    }
}

File: app/src/main/java/com/mb/catsmoker/RootLSPosedActivity.kt
package com.mb.catsmoker

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.button.MaterialButton
import java.io.File

class RootLSPosedActivity : AppCompatActivity() {

    companion object {
        private val packageProps = mapOf(
            "com.YoStar.AetherGazer" to createOP12Props(),
            // ... (keep all other package mappings)
            "vng.games.revelation.mobile" to createOP12Props()
        )

        private fun createOP12Props(): Map<String, String> = mapOf(
            "MANUFACTURER" to "OnePlus",
            "MODEL" to "PJD110",
            "BRAND" to "OnePlus",
            "DEVICE" to "OnePlus12"
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_root_lsposed)

        findViewById<MaterialButton>(R.id.btnActivate).setOnClickListener {
            activateSpoofing()
        }
    }

    private fun activateSpoofing() {
        if (!isRooted()) {
            Toast.makeText(this, "Root access not detected. Please ensure your device is rooted.", Toast.LENGTH_LONG).show()
            return
        }

        try {
            spoofDeviceProperties()
            Toast.makeText(this, "Game Unlocker Activated - Spoofed as OnePlus 12", Toast.LENGTH_LONG).show()
        } catch (e: Exception) {
            Toast.makeText(this, "Spoofing failed: ${e.message}", Toast.LENGTH_LONG).show()
        }
    }

    private fun isRooted(): Boolean {
        val paths = arrayOf(
            "/system/bin/su",
            "/system/xbin/su",
            "/sbin/su",
            "/data/local/xbin/su",
            "/data/local/bin/su",
            "/system/sd/xbin/su",
            "/system/bin/failsafe/su"
        )
        return paths.any { File(it).exists() } || try {
            Runtime.getRuntime().exec("which su").waitFor() == 0
        } catch (e: Exception) {
            false
        }
    }

    private fun spoofDeviceProperties() {
        packageProps.forEach { (_, props) ->
            props.forEach { (key, value) ->
                try {
                    val field = android.os.Build::class.java.getDeclaredField(key)
                    field.isAccessible = true
                    field.set(null, value)
                    field.isAccessible = false
                } catch (e: Exception) {
                    // Log failure but continue with other properties
                }
            }
        }
    }
}